docs/Contributing/Blueprints/Fleet-Installers.md

```diff
@@ -0,0 +1,431 @@
+# Fleet Installers
+
+## Goal
+
+[#5757](https://github.com/fleetdm/fleet/issues/5757)
+
+```
+As a user, I want to be able to download a Fleet-osquery installer (aka Orbit) in the Fleet UI
+so that I can add hosts to Fleet without having to know how to successfully generate an
+installer with the `fleetctl package` command.
+
+Figma wireframes: https://www.figma.com/file/hdALBDsrti77QuDNSzLdkx/?node-id=6740%3A267448
+```
+
+## Command `fleetctl package`
+
+Currently users use the `fleetctl package` command to generate Fleet-osquery packages.
+
+The `fleetctl package` command has the following *required* configuration that is specific to a "Fleet Deployment":
+- `--fleet-url`: The URL that the hosts will use to connect to the Fleet server.
+- `--enroll-secret`: Global or team enroll secret to use when enrolling the host to Fleet.
+
+As the goal states, we would like to provide functionality in Fleet to automatically generate and download such packages from the UI.
+
+## How
+
+We can implement such functionality in two ways:
+
+- Option A. Fleet Server to generate such packages itself.
+- Option B. Separate "Packager" service.
+
+There's a lot of platform specific logic and tooling involved in packaging, and one of Fleet's primary goals is to keep deployment/infrastructure simple for On-Prem deployments.
+To that end, we believe the best option is Option B, implementing the functionality as a separate service.
+
+## Packager Service
+
+The "Fleet Packager" service will implement all the package generation logic. Think of it like offering the `fleetctl package` command functionality but as a REST service.
+
+```mermaid
+graph LR;
+    A[User-Agent/<br>Browser]-- Fleet API -->B[Fleet<br>UI/Server]
+    A-- Packager<br>API -->C
+    subgraph FleetDM Hosted Infrastructure
+        direction LR
+        subgraph <br>pkg.fleetctl.com
+            direction TB;
+            C[Packager<br>Service]
+            packages[(Generated<br>packages)]
+        end
+        direction TB;
+        D[tuf.fleetctl.com]    
+        C-- Fetch<br>Targets -->D;
+    end
+    E[Apple's <br> Notary <br> Server]
+    C-- Notary<br>API ---->E
+```
+
+Configurations:
+- The `Fleet UI/Server` will allow configuring the "Fleet Packager" URL (default value being FleetDM Hosted Packager service, something like `pkg.fleetctl.com`).
```

Comment:
I believe the goal _is_ to expose the service publicly, so that any Fleet user (On-Prem) can make use of it w/o requiring changes on their configuration/deployment. (In a similar fashion to how we offer https://tuf.fleetctl.com/ for everyone)